shm/src/map.cpp

```diff
@@ -0,0 +1,44 @@
+#include "map.hpp"
+
+#include <algorithm>
+#include <random>
+#include <set>
+#include <utility>
+
+constexpr size_t kIslandAmt = 10;
+constexpr size_t kMapWidth = 50;
+constexpr size_t kMapHeight = 50;
+
+Map::Map() {
+    generateIslands();
+}
+
+Island* Map::getIsland(const Coordinates& coordinates) {
+    auto island = std::find_if(islands_.begin(), islands_.end(),
+                               [&](const auto& island) {
+                                   return coordinates == island.getPosition();
+                               });
+    return island != std::end(islands_) ? &*island : nullptr;
+}
+
+void Map::generateIslands() {
+    std::random_device rd;
+    std::mt19937 eng(rd());
+    islands_.reserve(kIslandAmt);
```

Comment:
You can use: https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution
As described in the task.